src/anndata/_core/sparse_dataset.py

```diff
@@ -569,21 +572,29 @@ def _to_backed(self) -> BackedSparseMatrix:
         mtx = format_class(self.shape, dtype=self.dtype)
         mtx.data = self.group["data"]
         mtx.indices = self.group["indices"]
-        mtx.indptr = self.indptr
+        mtx.indptr = self._indptr
         return mtx
 
-    def to_memory(self) -> ss.csr_matrix | ss.csc_matrix:
+    def to_memory(self) -> spmatrix | SpArray:
+        """Returns an in-memory representation of the sparse matrix.
+
+        Returns
+        -------
+            The in-memory representation of the sparse matrix.
+        """
         format_class = get_memory_class(self.format)
```

Comment:
This never returns `csr_array`/`csc_array`, should it?

Reply:
At the moment, it shouldn't.  But we need at add a setting in for that, thanks for the reminder.   Will PR now

Reply:
OK, then that doesn’t stop this PR